pypeit/core/arc.py

```diff
@@ -916,25 +916,27 @@ def detect_lines(censpec, sigdetect=5.0, fwhm=4.0, fit_frac_fwhm=1.25, input_thr
 
     # TODO: Why is this here? Can't the calling function be required to
     # pass a single spectrum?  This is not reflected in the docstring.
-    if len(censpec.shape) == 3:
-        detns = censpec[:, 0].flatten()
-    else:
-        detns = censpec.copy()
-    detns = detns.astype(np.float)
-    xrng = np.arange(detns.size, dtype=np.float)
+    # OLD CODE
+    #if len(censpec.shape) == 3:
+    #    detns = censpec[:, 0].flatten()
+    #else:
+    #    detns = censpec.copy()
+    #detns = detns.astype(np.float)
+
+    bpm_out = censpec == 0.0 if bpm is None else bpm
+    xrng = np.arange(censpec.size, dtype=np.float)
```

Comment:
Is there a case where the calling function passes a 3D array?  If this works, I'd suggest removing the commented code and the TODO comment above.